### PR TITLE
Use the unencoded URL

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request

* This broke the WPCC flows

#### Testing instructions

- Open an incognito window
- Go to gravatar.com
- Click create your own Gravatar
- Be taken to wordpress.com/start
- Change https://wordpress.com ot http://calypso.localhost:3000
- Create an account
- Check you are redirected back to Gravatar.

### CODE REVIEW

File: client/signup/config/flows.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function getPostsDestination( dependencies ) {
 function getRedirectDestination( dependencies ) {
 	if (
 		dependencies.oauth2_redirect &&
-		dependencies.oauth2_redirect.match( /^https%3A%2F%2Fpublic-api.wordpress.com/i )
+		dependencies.oauth2_redirect.match( /^https:\/\/public-api.wordpress.com/i )
```
Comment (blowery) [inline client/signup/config/flows.js]: shouldn't the periods in the url be escaped with `\`?

Comment (blowery) [reply client/signup/config/flows.js]: might be safer to use `.startsWith()` instead of match

**Comment by fditrapani** [comment]
I tested it out and it worked as described.

File: client/signup/config/flows.js (hunk 1 above)
Comment (scruffian) [reply client/signup/config/flows.js]: Updated to use this approach.

File: client/signup/config/flows.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ function getPostsDestination( dependencies ) {
 function getRedirectDestination( dependencies ) {
 	if (
 		dependencies.oauth2_redirect &&
-		dependencies.oauth2_redirect.match( /^https%3A%2F%2Fpublic-api.wordpress.com/i )
+		startsWith( dependencies.oauth2_redirect, 'https://public-api.wordpress.com' )
```
Comment (blowery) [inline client/signup/config/flows.js]: Cool. If you're going to use lodash's startsWith, you don't need the existence check on the line above. `String` also has a native `startsWith` method that we polyfill.

Comment (scruffian) [reply client/signup/config/flows.js]: Ah ok cool, I wasn't sure if we had the polyfill...

Comment (scruffian) [reply client/signup/config/flows.js]: Using the native function now, for performance :)

Comment (blowery) [reply client/signup/config/flows.js]: Looks great!

**Pull request merged**